### ISSUE TITLE
[REQUEST]: Application logs add "Elastic Agent" option

### ISSUE THREAD

**Issue by SylvainJuge** [elastic/observability-docs#2607]
### Description

With the new section on the Application Logs, we document the ingestion strategies that mostly rely on the semi-deprecated Filebeat.

We need to provide documentation how to achieve the same with a Fleet integration, maybe with the "Custom Logs" + some configuration for now.

Also, we could take the opportunity to re-structure a bit this guide by relying on the log format.
- Plain-text logs
  - Changes required in app/config
  - Filebeat/Fleet configuration to ingest those logs (both would require an ingest pipeline).
- ECS formatted logs
  - with ECS Logging library
  - with ECS Reformatting (and thus APM agent)
  - Filebeat/Fleet configuration to ingest those logs
- Log sending

### Resources

current Application Logs doc: https://www.elastic.co/guide/en/observability/current/application-logs.html

### Collaboration

TBD. The docs and product team will work together to determine the best path forward.

### Point of contact.

**Main contact:** @SylvainJuge 

**Stakeholders:**

**Comment by SylvainJuge** [comment]
Initialy mentioned here : https://github.com/elastic/apm/issues/765#issuecomment-1425759060

**Comment by felixbarny** [comment]
I hope we'll have a dedicated ECS integration for Elastic Agent at some point. I've prototyped one (https://github.com/elastic/integrations/pull/2972) but it has some blockers. There seems to be movement on the blockers, though. Still, in the meantime it probably makes sense to document how to use the custom logs integration. The config would look very similar to the Filebeat config, just copy/pasted to the advanced section in the Elastic Agent integration.

**Comment by SylvainJuge** [comment]
@felixbarny do you know if there was any progress on the blockers you mentioned for a proper integration ?

Until we have a ready to use integration, I think it would still make sense to document using Elastic Agent based log ingestion, even if that requires a bit more effort than it would through a dedicated integration. The main issue I see for now is that people using Elastic Agent might think it's not supported, whereas it's just an alternate ingestion tool.

**Comment by felixbarny** [comment]
Yes, the blockers have been removed by we still don't have a dedicated ECS logging integration. +1 to adding docs for how to use Elastic Agent as a shipper in the meantime.

**Comment by bmorelli25** [comment]
I've added EA application logging to our Logs+ planning board in https://github.com/elastic/observability-docs/issues/2954.

**Comment by bmorelli25** [comment]
Moving to high priority. We'll get this prioritized after Oct. 31st.

**Comment by SylvainJuge** [comment]
> Moving to high priority. We'll get working on this after Oct. 31st.

Good to know this is moving forward, if needed we have a few configuration examples that are available in the apm contrib repository: https://github.com/elastic/apm-contrib/tree/main/apm-agent-java/log-ingest, those are not "documentation level quality" but definitely relevant if needed to understand and experiment. Also we might add a new variant with the agent, or at least mention it once it's officially added to the documentation.

**Comment by bmorelli25** [comment]
The plan right now is to doc both Filebeat and Elastic Agent for self-managed/ESS users. We can't get Filebeat to work with Serverless (yet) so we'll only doc Elastic Agent with Serverless.

**Comment by bmorelli25** [comment]
Update: Filebeat works but will be much easier to use with Serverless in 8.11 because of this PR: https://github.com/elastic/beats/pull/36649. We plan on doc'ing EA & Filebeat in both our classic docs and our Serverless docs.
